src/ipld/dag_cbor.rs

```diff
@@ -4,6 +4,7 @@ use crate::ipld::{BlockError, Ipld, IpldError};
 use byteorder::{BigEndian, ByteOrder};
 use cid::Cid;
 use std::{
+    cmp::Ordering,
```

Comment:
```suggestion
```

If/when you simplify the if-elseif-else away with my suggestion, this becomes unnecessary and will most likely fail compilation.